File: go/17_http.go
package main

import (
	"encoding/json"
	"fmt"
	"net/http"
	"time"
)

func main() {
	client := &http.Client{Timeout: 5 * time.Second}
	resp, err := client.Get("https://httpbin.org/get")
	if err != nil {
		fmt.Println("error:", err)
		return
	}
	defer resp.Body.Close()

	fmt.Println("status:", resp.StatusCode)

	var data map[string]interface{}
	json.NewDecoder(resp.Body).Decode(&data)
	fmt.Println("json keys:", data)
}

File: python/17_http.py
# 17 http / api
# needs: pip install requests

try:
    import requests
    r = requests.get("https://httpbin.org/get", timeout=5)
    print("status:", r.status_code)
    print("json keys:", list(r.json().keys()))
except ImportError:
    print("install requests: pip install requests")
except Exception as e:
    print("error:", e)
